sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -295,13 +289,8 @@ class Dataset[T] private[sql](
     // first `truncate-3` and "..."
     schema.fieldNames.map(SchemaUtils.escapeMetaCharacters).toSeq +: data.map { row =>
       row.toSeq.map { cell =>
-        val str = cell match {
-          case null => "NULL"
-          case binary: Array[Byte] => binary.map("%02X".format(_)).mkString("[", " ", "]")
-          case _ =>
-            // Escapes meta-characters not to break the `showString` format
-            SchemaUtils.escapeMetaCharacters(cell.toString)
-        }
+        // Escapes meta-characters not to break the `showString` format
+        val str = SchemaUtils.escapeMetaCharacters(cell.toString)
```

Comment:
This could potentially throw NullPointerException if `cell` is null. This was handled in the original code with match-case statement. 

We could replace it with `s"$cell"` or handle null separately.

Reply:
This should never be null as `ToPrettyString.nullable` is false. I can add an assert to enforce it.